### ISSUE TITLE
💡 Support for Hypixel's ratelimits

### ISSUE THREAD

**Issue by ixnoahlive** [NoahTheNerd/lavacake#5]
**Does this suggestion solve an existing problem? If yes, what is said problem?**
It solves the issue where ratelimits from Hypixel are poorly/not at all handled by Lavacake.

**Provide information about your suggestion**
It should in some way control/queue player requests to optimize the density of player requests.
Otherwise, some more carefulness with ratelimits would be beneficial

**How important would this be? Would it be quality of life or is it absolutely essential?**
Very needed, but not the highest of priorities.

**Comment by NoahTheNerd** [comment]
Added High Priority. Is not top priority but 100% worth looking into.

**Comment by NoahTheNerd** [comment]
New proposal for a resolution is that requests are spread across the time, so if you got a limit of 300 requests per 5 minutes it'll add it to a queue & then send one every second. Would like thoughts on this from contributors.

**Comment by robere2** [comment]
> New proposal for a resolution is that requests are spread across the time, so if you got a limit of 300 requests per 5 minutes it'll add it to a queue & then send one every second. Would like thoughts on this from contributors. (which at the minute is just @robere2 and I)

I agree that this is a good solution. The timings should be based on the `RateLimit` headers if possible. If not or if that's something you want to tackle in the future, they should at least be configurable.
